### PR TITLE
Migrate use case docs into Sphinx

### PR DESCRIPTION
**Pull request by gnidan**
- Start with pandoc-converted RST file
- Add Solidity extensions to project for highlighting
- Update all code examples, either:
  - Directly referencing file in repo via `literalinclude` directive
  - Providing necessary preprocessing command whose output is pasted into RST source
- Replace `*<glossary-term>*` with glossary reference.
- Remove use cases from README.md

ℹ️ [Preview](http://ethpm.github.io/ethpm-spec/contrib/ethpm/use-cases-docs/)

### CODE REVIEW

File: docs/source/use-cases.rst [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+optional data that can be included. However, for the remainder of our
+examples we will be using minimalistic Packages to keep our examples as
+succinct as possible.
+
+.. _package-with-an-inheritable-contract-and-a-dependency:
+
+Package with an Inheritable Contract and a Dependency
+-----------------------------------------------------
+
+Now that we've seen what a simple package looks like, lets see how to
```
Comment (njgheorghita) [inline docs/source/use-cases.rst]: nitpick typo - `let's see how dependencies ...`

File: docs/source/use-cases.rst [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+             "...": "..."
+           }
+         }
+       }
+     }
+   }
+
+
+While it is not required to include the contract ABI and NatSpec
+information, it does provide those using this package with they data
```
Comment (njgheorghita) [inline docs/source/use-cases.rst]: nitpick typo - `with (they) data they would need ...`

File: docs/source/use-cases.rst [hunk 3]
```diff
@@ -0,0 +1,718 @@
...
+     }
+   }
+
+
+The first thing to point out is that unlike our ``standard-token``
+contract, we've included the ``bytecode``, ``runtime_bytecode`` and
+``compiler`` information in the ``SafeMathLib`` section of the
+``contract_type`` definition. This is because we are also including a
+deployed instance of this contract and need to require adequate
+information for package managers to verify that the contract sound at
```
Comment (njgheorghita) [inline docs/source/use-cases.rst]: nitpick typo - sound -> found

**Comment by njgheorghita** [comment]
Other than a couple typos that we might as well iron out here - looks great. Though, personally the use cases are much too wordy/difficult to parse for my liking. 
We have two descriptors for a use case-
1. actual package name - i.e. `owned` - short & succinct but not descriptive
2. use case def - i.e. `Package with a contract with link dependencies on a contract from a package dependency` - descriptive but long and difficult to parse mentally

I think it'd be great if we could either improve (i.e. shorten but maintain descriptiveness) 2. defs - or come up with something in-between 1. and 2. Though, i'm certain once we get plantuml visuals in here - those will do a lot to help cut through the wordiness of it all.

**Comment by gnidan** [comment]
Thanks @njgheorghita @pipermerriam 

Piper– I opened an issue for updating the packages. I figure we'll do that once this spec is more finalized for v2.

**Comment by gnidan** [comment]
Ah @njgheorghita re your other point:
> Other than a couple typos that we might as well iron out here - looks great. Though, personally the use cases are much too wordy/difficult to parse for my liking. 
> We have two descriptors for a use case-
> 1. actual package name - i.e. `owned` - short & succinct but not descriptive
> 2. use case def - i.e. `Package with a contract with link dependencies on a contract from a package dependency` - descriptive but long and difficult to parse mentally
> 
> I think it'd be great if we could either improve (i.e. shorten but maintain descriptiveness) 2. defs - or come up with something in-between 1. and 2. Though, i'm certain once we get plantuml visuals in here - those will do a lot to help cut through the wordiness of it all.

This is valid, and I was a bit frustrated by the lack of clarity myself. I'll fix that up in another PR.
